src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -346,7 +346,7 @@ public IEnumerable<FileInfo> LoadInventory(FileInfo file)
                 file.Refresh();
                 if (!file.Exists)
                 {
-                    throw new ArgumentException("Inventory file should exist.");
```

Comment:
ArgumentException didn't account for rebuild scenarios. This infrastructure is likely getting complex enough to warrant its own tests.